ols/app/endpoints/feedback.py

```diff
@@ -1,19 +1,111 @@
 """Handler for REST API call to provide user feedback."""
 
+import json
 import logging
+from pathlib import Path
 from typing import Any
 
 from fastapi import APIRouter, Depends
 
-from ols.app.models.models import FeedbackRequest, FeedbackResponse
+from ols.app.models.models import (
+    FeedbackRequest,
+    FeedbackResponse,
+    RemoveFeedbackRequest,
+)
+from ols.utils import config
 from ols.utils.auth_dependency import auth_dependency
 
 logger = logging.getLogger(__name__)
 router = APIRouter(prefix="/feedback", tags=["feedback"])
 
 
+def list_feedbacks() -> list[str]:
+    """List feedbacks in the local filesystem.
+
+    Returns:
+        List of feedback files (without the ".json" extension).
+    """
+    storage_path = Path(config.ols_config.feedback_storage_location)
+    feedback_files = list(storage_path.glob("*.json"))
+    # extensions are trimmed, eg. ["12345678-abcd-0000-0123-456789abcdef", ...]
+    feedbacks = [f.stem for f in feedback_files]
+    logger.info(f"'{len(feedbacks)}' feedbacks found")
+    return feedbacks
+
+
+def store_feedback(conversation_id: str, user_id: str, feedback: dict) -> None:
+    """Store feedback in the local filesystem.
+
+    Args:
+        conversation_id: The conversation ID (UUID).
+        user_id: The user ID (UUID).
+        feedback: The feedback object.
+    """
+    # ensures storage path exists
+    storage_path = Path(config.ols_config.feedback_storage_location)
+    if not storage_path.exists():
+        storage_path.mkdir(parents=True)
+
+    data_to_store = {
+        "metadata": {
+            "conversation_id": conversation_id,
+            "user_id": user_id,
+        },
+        "feedback": feedback,
+    }
+
+    # stores feedback in a file named after the conversation id
+    feedback_file_path = storage_path / f"{conversation_id}.json"
```

Comment:
this isn't sufficiently unique to support cases where a user provides feedback on multiple interactions within a single conversation.  

or cases where someone submits feedback multiple times for the same interaction (less likely, but no particular reason we shouldn't allow it).

so i think this requires another identifier, just a uuid generated for each feedback submission.

Reply:
replaced for uuid